src/main/lws/lws-service.js

```diff
@@ -131,14 +131,22 @@ export class LWSService {
 		}
 	}
 
-	async genSignature(nonce, publicKey, conn) {
-		const check = this.checkWallet(publicKey, conn);
-		if (!check.unlocked) {
+	genSignature(nonce, publicKey, privateKey) {
+		try {
+			const msgHash = ethUtil.hashPersonalMessage(Buffer.from(nonce, 'hex'));
+			const signature = ethUtil.ecsign(msgHash, Buffer.from(privateKey, 'hex'));
+			return signature;
+		} catch (error) {
+			log.error(error);
 			return null;
 		}
-		const msgHash = ethUtil.hashPersonalMessage(Buffer.from(nonce, 'hex'));
-		const signature = ethUtil.ecsign(msgHash, Buffer.from(check.privateKey, 'hex'));
-		return signature;
+	}
```

Comment:
is this needed?  In the implementation I have on local I'm able to just do JSON.stringify to the original return signature result and it works correctly with the verification on the server side

Reply:
Yes, you are transferring stringified Buffer, it will work only between node projects. We don't know which server tech our customers will use.